nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -312,100 +342,236 @@ func (w *deploymentWatcher) StopWatch() {
 	w.exitFn()
 }
 
-// watch is the long running watcher that takes actions upon allocation changes
+// watch is the long running watcher that watches for both allocation and
+// deployment changes. Its function is to create evaluations to trigger the
+// scheduler when more progress can be made, to fail the deployment if it has
```

Comment:
What does `when more progress can be made` mean in this case?

Reply:
Added more to the comment